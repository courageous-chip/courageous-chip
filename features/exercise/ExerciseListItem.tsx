import { gql } from "@apollo/client";
import React, { FC } from "react";
import { Text, VirtualizedListProps } from "react-native";

import { ExerciseListItemFields } from "./__generated__/ExerciseListItemFields";

export const EXERCISE_LIST_ITEM_FIELDS_FRAGMENT = gql`
  fragment ExerciseListItemFields on Exercise {
    id
    name
  }
`;

export const keyExtractor: VirtualizedListProps<
  ExerciseListItemFields
>["keyExtractor"] = function ({ id }) {
  return id;
};

export const renderItem: VirtualizedListProps<
  ExerciseListItemFields
>["renderItem"] = function ({ item: { name } }) {
  return <ExerciseListItem name={name} />;
};

type Props = Pick<ExerciseListItemFields, "name">;

export const ExerciseListItem: FC<Props> = function ({ name }) {
  return <Text>{name}</Text>;
};
