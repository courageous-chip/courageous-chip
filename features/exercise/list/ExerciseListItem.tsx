import { gql } from "@apollo/client";
import React, { FC } from "react";
import { FlatListProps, StyleSheet, Text, View } from "react-native";

import { ExerciseListItemFields } from "./__generated__/ExerciseListItemFields";

export const EXERCISE_LIST_ITEM_FIELDS_FRAGMENT = gql`
  fragment ExerciseListItemFields on Exercise {
    id
    name
  }
`;

export const ItemSeparatorComponent: FlatListProps<
  ExerciseListItemFields
>["ItemSeparatorComponent"] = function () {
  return <View style={styles.itemSeparator} />;
};

export const keyExtractor: FlatListProps<
  ExerciseListItemFields
>["keyExtractor"] = function ({ id }) {
  return id;
};

export const renderItem: FlatListProps<
  ExerciseListItemFields
>["renderItem"] = function ({ item: { name } }) {
  return <ExerciseListItem name={name} />;
};

type Props = Pick<ExerciseListItemFields, "name">;

export const ExerciseListItem: FC<Props> = function ({ name }) {
  return (
    <View style={styles.itemContainer}>
      <Text style={styles.itemText}>{name}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  itemContainer: { justifyContent: "center", minHeight: 64 },
  itemSeparator: { backgroundColor: "#f7f7f7", height: 1, width: "100%" },
  itemText: { marginHorizontal: 20 },
});
