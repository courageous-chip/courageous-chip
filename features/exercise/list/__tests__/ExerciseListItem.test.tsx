import React from "react";
import renderer from "react-test-renderer";

import { ExerciseListItem } from "../ExerciseListItem";

describe("<ExerciseListItem />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ExerciseListItem id="1" name="🏋️‍♀️ Deadlift" onPress={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
