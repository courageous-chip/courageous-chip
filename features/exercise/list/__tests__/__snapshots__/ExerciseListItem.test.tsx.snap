// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExerciseListItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "minHeight": 64,
    }
  }
>
  <Text
    style={
      Object {
        "marginHorizontal": 20,
      }
    }
  >
    🏋️‍♀️ Deadlift
  </Text>
</View>
`;
